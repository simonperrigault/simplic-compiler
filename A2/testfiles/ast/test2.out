Program
  List
    FuncDecl
      IdDecl ID="main"
      List
      Block
        List
          DeclStmt
            List
              Decl
                IdDecl ID="x"
                Opt
          DeclStmt
            List
              Decl
                IdDecl ID="y"
                Opt
                  Numeral NUMERAL="2"
          AssignStmt
            IdDecl ID="x"
            Numeral NUMERAL="3"
          AssignStmt
            IdDecl ID="x"
            IdUse ID="y"
