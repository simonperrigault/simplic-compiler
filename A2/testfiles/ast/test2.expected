Program
  List
    FuncDecl
      IdDecl ID="main"
      List
      List
        Decl
          IdDecl ID="x"
          Opt
        Decl
          IdDecl ID="y"
          Opt
            Numeral NUMERAL="2"
        Assignment
          IdDecl ID="x"
          Numeral NUMERAL="3"
        Assignment
          IdDecl ID="x"
          IdUse ID="y"